src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.cpp

```diff
@@ -60,28 +51,18 @@ Scalar<DataType> spatial_weight_function(
 template <size_t SpatialDim, typename Frame, typename DataType>
 void spacetime_deriv_of_spatial_weight_function(
     const gsl::not_null<tnsr::a<DataType, SpatialDim, Frame>*> d4_weight,
-    const tnsr::I<DataType, SpatialDim, Frame>& coords,
-    const double sigma_r) noexcept {
+    const tnsr::I<DataType, SpatialDim, Frame>& coords, const double sigma_r,
+    const Scalar<DataType>& weight_function) noexcept {
   if (UNLIKELY(get_size(get<0>(*d4_weight)) != get_size(get<0>(coords)))) {
     *d4_weight = tnsr::a<DataType, SpatialDim, Frame>(get_size(get<0>(coords)));
   }
-  const DataType pre_factor =
-      get(spatial_weight_function(coords, sigma_r)) * (-2. / pow<2>(sigma_r));
-  // time derivative of weight function is zero
-  get<0>(*d4_weight) = 0.;
+  // use 0th component to avoid allocations
+  get<0>(*d4_weight) = get(weight_function) * (-2. / pow<2>(sigma_r));
```

Comment:
while you're fixing things on this line `pow<2>`->`square`

Reply:
Agreed!